registry/handlers/manifests.go

```diff
@@ -476,3 +712,12 @@ func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Reques
 
 	w.WriteHeader(http.StatusAccepted)
 }
+
+// annotatedTag will annotate OCI tags by prepending a string, and leave docker
+// tags unmodified.
+func (imh *manifestHandler) annotatedTag(oci bool) string {
```

Comment:
Yea, this is a big no. Sorry, but this is not at all the right path here.

OCI images should be treated as any other.